pos_tare/static/src/js/screens.js

```diff
@@ -0,0 +1,202 @@
+                if (this.tare > 0.0) {
+                    var order = this.pos.get_order();
+                    var orderline = order.get_last_orderline();
+                    orderline.set_tare(this.tare, false);
+                }
+            }
+        },
+
+        // /////////////////////////////
+        // Custom Section
+        // /////////////////////////////
+        get_product_gross_weight_string: function () {
+            var product = this.get_product();
+            var defaultstr = (this.gross_weight || 0).toFixed(3) + ' Kg';
+            if (!product || !this.pos) {
+                return defaultstr;
+            }
+            var unit_id = product.uom_id;
+            if (!unit_id) {
+                return defaultstr;
+            }
+            var unit = this.pos.units_by_id[unit_id[0]];
+            var weight = round_pr(this.gross_weight || 0, unit.rounding);
+            var weightstr = weight.toFixed(
+                Math.ceil(Math.log(1.0/unit.rounding) / Math.log(10) ));
+            weightstr += ' ' + unit.name;
+            return weightstr;
+        },
+
+        onchange_tare: function () {
+            this.tare = this.check_sanitize_value('#input_weight_tare');
+            this.set_weight(this.gross_weight);
+        },
+
+        onchange_gross_weight: function () {
+            var gross_weight = this.check_sanitize_value('#input_gross_weight');
+            this.set_weight(gross_weight);
+        },
+
+        check_sanitize_value: function (input_name) {
+            var res = this.$(input_name)[0].value.replace(',', '.').trim();
+            if (isNaN(res)) {
+                this.$(input_name).css("background-color", "#F66");
+                return undefined;
+            }
+            this.$(input_name).css("background-color", "#FFF");
+            return parseFloat(res, 10);
+        },
+
+    });
+
+    screens.PaymentScreenWidget.include({
+        validate_order: function(options) {
+            var order = this.pos.get_order();
+            var orderlines = Array.from(order.get_orderlines());
+
+            if (orderlines.some(leq_zero_qty)) {
+                var _super_validate_order = this._super.bind(this);
+                var wrong_orderline = orderlines.find(leq_zero_qty);
+                var wrong_product = wrong_orderline.get_product().display_name;
+                this.gui.show_popup('confirm', {
+                    title: _t('Quantity lower or equal to zero'),
+                    body:  _.str.sprintf(
+                        _t("The quantity for \"%s\" is lower or equal to" +
+                        " zero. Call for help unless you're perfectly" +
+                        " sure you are doing right."), wrong_product),
+                    confirm: function() {
+                        _super_validate_order();
+                    },
+                });
+                return;
+            }
+            return this._super(options);
+        },
```

Comment:
:tada:

Reply:
Glad you liked it! Thank you for the review :)